### PR TITLE
Fixed flycheck-rust-cargo-has-command-p for newer Cargo versions

### LINKED ISSUE #1916: `flycheck-rust-cargo-has-command-p` returns `nil` incorrectly
Thank you for taking the time to report an issue and improve Flycheck.  This template is for **actual bugs you observed**.  If you have trouble setting up Flycheck, or if you have a question, please use the relevant issue template instead.

## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description
Running `(flycheck-rust-cargo-has-command-p "run")` (or any other Cargo command) returns `nil` when it should return `t`. This is because `flycheck-rust-cargo-has-command-p` uses `cargo --list` as a basis, which returns descriptions for each subcommand (ever since https://github.com/rust-lang/cargo/pull/9848). This means `(member command ...)` returns `nil`.

## Steps to reproduce
1. Run `(flycheck-rust-cargo-has-command-p "run")` with a sufficiently recent (>= 1.56) version of Cargo installed.
2. Observe that it returns `nil`

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior
The command should return `t` when the command is installed

Emacs configuration:
- [ ] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [x] Doom Emacs
- [ ] Other shared configuration

[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

### PR DESCRIPTION
**Pull request by PatrickNorton**
Should fix #1916.

### CODE REVIEW

**Comment by bergey** [comment]
This looks good to me.  I encountered the same bug today, and wrote an equivalent fix, but `string-prefix-p` is simpler & more efficient than what I wrote.

**Comment by rymndhng** [comment]
I've also run into this, and this fix works for me.

**Comment by gerlacdt** [comment]
Patch works for me too, thx!

**Comment by ngoldbaum** [comment]
I just ran into this and would appreciate it if this could be merged.

**Comment by Gastove** [comment]
Also just smacked into this. It seems like we've got a couple decent solutions; @cpitclaudel, any chance you or another maintainer could give this a review?

**Comment by cpitclaudel** [comment]
LGTM, but Flycheck uses `seq`, not `cl-lib`, so it would be a bit better to use `seq-find`.  Additionally, if `cargo --list` prints `foo-blah`, won't the new code conclude that `foo` is available?

**Comment by ngoldbaum** [comment]
maybe #1948 is a better fix?

**Comment by PatrickNorton** [comment]
I've changed the code to use `seq-find`, but I think that #1948 does have a more robust solution overall—I hadn't thought about what would happen if one command were to be a substring of another.

File: flycheck.el [hunk 1]
```diff
@@ -11431,8 +11431,9 @@ versions inferior to 1.25)."
 
 Execute `cargo --list' to find out whether COMMAND is present."
   (let ((cargo (funcall flycheck-executable-find "cargo")))
-    (member command (mapcar #'string-trim-left
-                            (ignore-errors (process-lines cargo "--list"))))))
+    (seq-find (lambda (x) (string-prefix-p command x))
+              (mapcar #'string-trim-left
+                      (ignore-errors (process-lines cargo "--list"))))))
```
Comment (cpitclaudel) [inline flycheck.el:11436]: Would something like this work?

```suggestion
    (member command 
      (mapcar (lambda (line)
                (replace-regexp-in-string "\\s-*\\(\\S-+\\).*\\'" "\\1" line))
              (ignore-errors (process-lines cargo "--list"))))))
```

**Comment by cpitclaudel** [comment]
Unfortunately the code in #1948 will corrupt the match data.  I've posted an untested piece of code that might do the trick; can someone check?  Thanks!

**Comment by cpitclaudel** [comment]
Thanks!  I'll wait for one more person to approve this or a few days to pass and then I'll merge it.  Sorry for letting this sit so long and thanks a lot for being more reactive than me.

**Comment by PatrickNorton** [comment]
No worries! Waiting for one more approval sounds fine. You happened to catch me during some down time; I wish I was usually as responsive as I am today :).

**Pull request merged**